django/contrib/admin/helpers.py

```diff
@@ -116,10 +117,14 @@ def __init__(
 
     @property
     def media(self):
-        if "collapse" in self.classes:
-            return forms.Media(js=["admin/js/collapse.js"])
         return forms.Media()
 
+    @cached_property
+    def is_collapsible(self):
+        if any([field in self.fields for field in self.form.errors]):
+            return False
+        return "collapse" in self.classes
```

Comment:
@sarahboyce @nessita I have considered this change several times and ultimately decided not to put this check in Python. I felt a UI-specific check would be better made in the template.

If you look at `fieldset.html`, you'll see on lines 3 and 39 that there is a check for the fieldset name, a prerequisite for the is_collapsible check.

If we move this check to Python, we can simplify the aforementioned if-statements in the template. I'm all for it if you are also OK with this since the nested if-statement made the template slightly more complicated to understand for those unaware of this bug.